src/Main.java

```diff
@@ -0,0 +1,173 @@
+        return name;
+    }
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public int getLevel() {
+        return level;
+    }
+    public void setLevel(int level) {
+        this.level = level;
+    }
+
+    public int getHp() {
+        return hp;
+    }
+    public void setHp(int hp) {
+        this.hp = hp;
+    }
+
+    public int getXp() {
+        return xp;
+    }
+    public void setXp(int xp) {
+        this.xp = xp;
+    }
+}
+
+
+// 4. Maak de volgende subklassen aan en laat deze voldoen aan de gestelde randvoorwaarden.
+// Subklasse: ElectricPokemon
+class ElectricPokemon extends Pokemon {
+    private int electricPower;
+    public ElectricPokemon(String name, int level, int hp, int xp) {
+        super(name, level, hp, xp);
+        this.electricPower = electricPower;
+    }
+    public void thunderPunch() {
+        System.out.println(getName() + " uses Thunder Punch!");
+    }
+    @Override
+    public void speak() {
+        System.out.println(getName() + " says, 'I'm electrifying!'");
+    }
+
+    // Getters and setters
+    public int getElectricPower() {
+        return electricPower;
+    }
+    public void setElectricPower(int electricPower) {
+        this.electricPower = electricPower;
+    }
+
+}
+
+
+// Subklasse: FirePokemon
+class FirePokemon extends Pokemon {
+    private int firePower;
+
+    public FirePokemon(String name, int level, int hp, int xp, int firePower) {
+        super(name, level, hp, xp);
+        this.firePower = firePower;
+    }
+    public void inferno() {
+        System.out.println(getName() + " uses Inferno!");
+    }
+    @Override
+    public void speak() {
+        System.out.println(getName() + " says, 'I'm on fire!'");
+    }
+
+    // Getters and setters
+    public int getFirePower() {
+        return firePower;
+    }
+    public void setFirePower(int firePower) {
+        this.firePower = firePower;
+    }
+
+
+}
+
+
+// Subklasse: WaterPokemon
+class WaterPokemon extends Pokemon {
+    private int waterPower;
+
+    public WaterPokemon(String name, int level, int hp, int xp, int waterPower) {
+        super(name, level, hp, xp);
+        this.waterPower = waterPower;
+    }
+    public void hydroPump() {
+        System.out.println(getName() + " uses Hydro Pump!");
+    }
+    @Override
+    public void speak() {
+        System.out.println(getName() + " says, 'I love water!'");
+    }
+
+    // Getters and setters
+    public int getWaterPower() {
+        return waterPower;
+    }
+    public void setWaterPower(int waterPower) {
+        this.waterPower = waterPower;
+    }
+
+}
+
+
+// Subklasse: GrassPokemon
+class GrassPokemon extends Pokemon {
+    private int leafPower;
+
+    public GrassPokemon(String name, int level, int hp, int xp, int leafPower) {
+        super(name, level, hp, xp);
+        this.leafPower = leafPower;
+    }
+    public void solarBeam() {
+        System.out.println(getName() + " uses Solar Beam!");
```

Comment:
Je geeft aan dat je niet zo goed weet hoe je het moet verbeteren. Laat me een suggestie doen: 

Je kunt in deze methode (en bij de "aanval methodes" van andre pokemons kun je het ook doen, maar probeer dan iets te varieren) met een if-statement checken wat de waarde van `leafPower` is en adhv die waarde bepalen wat er geprint wordt. Zoals dit: 

if (leafPower>50) {
sout(super.getName() + " uses a supercharged solar beam");
} else {
sout(super.getName() + " uses solar beam");
}

Reply:
Alternatief:
if (leafPower >0) {
leafPower--;
sout("do solar beam);
} else {
sout("out of leaf power");
}